src/main/scala/is/hail/methods/FilterIntervals.scala

```diff
@@ -1,33 +1,30 @@
 package is.hail.methods
 
-import is.hail.annotations.Annotation
-import is.hail.utils.{ArrayBuilder, Interval, IntervalTree}
-import is.hail.variant.{Locus, MatrixTable, Variant}
+import is.hail.utils.{Interval, IntervalTree}
+import is.hail.variant.MatrixTable
+import org.apache.spark.sql.Row
 
 import scala.collection.JavaConverters._
 
 object FilterIntervals {
   def apply(vsm: MatrixTable, intervals: java.util.ArrayList[Interval], keep: Boolean): MatrixTable = {
-    val pord = vsm.genomeReference.locusType.ordering
-    val iList = IntervalTree(vsm.genomeReference.locusType.ordering, intervals.asScala.toArray)
+    val iList = IntervalTree(vsm.locusType.ordering, intervals.asScala.toArray)
```

Comment:
I think it would be good to have a nice error message somewhere for the case where the type of the intervals given is not compatible with the row key of the matrix table.

Reply:
I was thinking of the case where someone has a matrix table with row key String such as gene name and they give an interval list containing variants. I don't know why someone would do that, but we probably want a nice error message if they do.

Reply:
You were totally right, there was no check.  I now call `requireRowKeyVariant` at the `FilterIntervals.apply` entry point from Python.